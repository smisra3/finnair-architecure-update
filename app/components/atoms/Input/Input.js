// @flow
import type { Node } from 'react';
import camelCase from 'lodash/camelCase';
import styled from 'styled-components';
import styles from './Input.style';
import FieldError from '../FieldError/FieldError';

type Props = {
  className: string,
  id: string | number,
  labelContent?: string,
  inheritedStyles?: string,
  errorMessage?: string,
  maxLength?: number,
  onKeyUp?: Function,
  iconClickHandler?: Function,
  inputRef?: Function,
  type?: string,
  placeholder?: string,
  value?: string,
  required?: boolean,
  dataSlnmId?: string,
  onChange?: Function
};

const keyUpHandler = (event, maxLength) => {
  if (event.target.value.length >= parseInt(maxLength, 10)) {
    // eslint-disable-next-line
    event.target.value = event.target.value.substring(0, parseInt(maxLength, 10));
  }
};

const Input = ({
  className,
  id,
  labelContent,
  errorMessage,
  inheritedStyles,
  maxLength,
  onKeyUp,
  iconClickHandler,
  inputRef,
  type,
  placeholder,
  value,
  onChange,
  required,
  dataSlnmId,
  labelAddOnComponent,
  ...others
}: Props): Node => {
  return (
    <div className={className}>
      <div className="label-wrapper">
        <label htmlFor={id}>{labelContent}</label>
        {labelAddOnComponent ? (<div className="addOnComponentWrapper">{labelAddOnComponent}</div>) : null}
      </div>
      <div className="input-wrapper">
        <input
          id={id}
          maxLength={maxLength}
          type={type}
          onKeyUp={(event) => {
            keyUpHandler(event, maxLength);
            if (onKeyUp) onKeyUp(event);
          }}
          onChange={(event) => {
            if (onChange) onChange(event);
          }}
          onBlur={(event) => {
            if (onBlur) onBlur(event);
          }}
          ref={inputRef}
          placeholder={placeholder}
          value={value}
          aria-invalid={errorMessage && errorMessage.length ? true : null}
          aria-required={required || null}
          data-slnm-id={camelCase(placeholder)}
          {...others}
        />
      </div>
      <FieldError errorMessage={errorMessage} dataSlnmId={dataSlnmId} />
    </div>
  )
}

Input.defaultProps = {
  type: 'text',
  labelContent: '',
  inheritedStyles: '',
  errorMessage: '',
  maxLength: 100,
  onKeyUp: () => { },
  iconClickHandler: () => { },
  inputRef: () => { },
  placeholder: '',
  value: '',
  required: false,
  dataSlnmId: 'InputField',
  onChange: () => { },
  onBlur: () => { },
  isParentSubmitted: false
};

export default styled(Input)`
  ${styles};
`;
export { Input as InputVanilla };
