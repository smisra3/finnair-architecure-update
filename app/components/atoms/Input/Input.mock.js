import { action } from '@storybook/addon-actions';
import defaultAnchor from '../Anchor/Anchor.mock';
import Anchor from '../Anchor/index';

const textInput = {
  className: 'example',
  labelContent: 'User Name',
  type: 'text',
  required: true,
  isError: false,
  onChange: action('changed'),
};

const checkboxInput = {
  className: 'example',
  labelContent: 'User Name',
  type: 'checkbox',
  required: true,
  isError: false,
  onChange: action('changed'),
  value: '',
};

const textInputWithCTA = {
  className: 'example',
  labelContent: 'Username',
  type: 'text',
  required: true,
  isError: false,
  onChange: action('changed'),
  labelAddOnComponent: (<Anchor {...defaultAnchor} >Forgot your username</Anchor>),
};


const textInputWithError = {
  className: 'example',
  labelContent: 'User Name',
  type: 'text',
  required: true,
  isError: true,
  errorMessage: 'Please enter your username or CRN',
  onChange: action('changed'),
};

export {
  textInput,
  textInputWithError,
  textInputWithCTA,
  checkboxInput,
};
