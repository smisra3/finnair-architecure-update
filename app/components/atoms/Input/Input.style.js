import { css } from 'styled-components';

const commonStyles = css`
  .label-wrapper {
    padding-bottom: 5px;
    width: 100%;
    position: relative;
    ${props => (props.labelAddOnComponent ? 'display:flex;' : '')};

    & label {
      height: 100%;
      font-weight: ${props => props.theme.fontBold};
      font-size: ${props => props.theme.fontSizeDefault};
      color: ${props => props.theme.textColorSecondary};
      margin: 0;
      ${props => (props.labelAddOnComponent ? 'flex: 1 1 auto;' : '')};
    }

    & .addOnComponentWrapper {
      flex: 0 0 auto;
      height: 100%;
      font-size: ${props => props.theme.fontSizeDefault};
    }
  }

  input {
    width: 100%;
    height: 18px;
    -webkit-tap-highlight-color: transparent;
    -webkit-appearance: none;
    -moz-appearance: none;
    border-radius: 0;
    font-size: ${props => props.theme.fontSizeBase};
    background-color: ${props => props.theme.backgroundColorPrimary};
    height: 100%;
    color: ${props => props.theme.textColorSecondary};
    padding: 7px 8px;
    border: 1px solid;
    box-shadow: none;
    border-color: ${props => (props.errorMessage ? props.theme.fieldBorderErrorColor : props.theme.fieldBorderColor)};
    &::placeholder {
      color: ${props => props.theme.textColorPlaceholder};
    }

    &::-ms-clear {
      display: none;
    }

    &:disabled {
      background: ${props => props.theme.backgroundColorDisabled};
      color: ${props => props.theme.textColorPlaceholder};
      border: 1px solid ${props => props.theme.fieldBorderDisabledColor};
    }

    &:focus {
      outline: none;
      border-color: ${props => (props.errorMessage ? props.theme.fieldBorderErrorColor : props.theme.fieldBorderFocusColor)};
    }
  }

  & .input-wrapper{
    margin-bottom: 10px;
  }

  input:-webkit-autofill {
    background-color: ${props => props.theme.fieldBorderColor};
  }
`;

export default commonStyles;
