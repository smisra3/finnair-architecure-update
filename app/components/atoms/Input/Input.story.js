// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  textInput,
  textInputWithCTA,
  textInputWithError,
} from './Input.mock';

// Import Vanilla Component to display markup and props
import { InputVanilla } from './Input';
// Import Styled Component to showcase variations
import Input from './index';

const wrapperStyle = {
  width: '400px'
};

storiesOf('Atoms', module).addWithChapters('Inputs', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => <InputVanilla {...textInput} className={`hide-default-sample ${textInput.className}`} />,
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Input Variations',
      sections: [
        {
          title: 'Default Text Box',
          sectionFn: () => <Input {...textInput} />,
        },
        {
          title: 'Text Box with CTA',
          sectionFn: () => <Input {...textInputWithCTA} />,
        },
        {
          title: 'Text Box With Inline Error',
          sectionFn: () => <Input {...textInputWithError} />,
        },
        {
          title: 'Text Box inside a wrapper of fixed width',
          sectionFn: () => (<div style={wrapperStyle}><Input {...textInputWithError} /></div>),
        },
      ],
    },
  ],
});
