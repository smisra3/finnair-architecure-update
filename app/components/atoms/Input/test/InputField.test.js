import { shallow } from 'enzyme';
import { InputVanilla as InputComponent } from '../Input';

/* eslint-disable max-len */
const props = {
  hideLabel: true,
  iconClass: 'iconClass',
  errorMessage: 'errorMessage',
  isError: true,
  onKeyUp: jest.fn(),
  onChange: jest.fn(),
  maxLength: 1,
  iconClickHandler: jest.fn(),
  dynamicPlaceholderLabel: true,
  placeholder: 'test',
  valid: true,
};

describe('Input Component', () => {
  let input = '';
  beforeEach(() => {
    input = shallow(<InputComponent {...props} />);
  });

  test('should render correctly', () => {
    expect(input).toMatchSnapshot();
  });

  test('should render correctly', () => {
    props.hideLabel = false;
    input = shallow(<InputComponent {...props} />);
    expect(input).toMatchSnapshot();
  });

  test('should render correctly', () => {
    props.iconClass = undefined;
    props.isError = false;
    input = shallow(<InputComponent {...props} />);
    expect(input).toMatchSnapshot();
  });

  test('should call keyup for add ', () => {
    const event = {
      target: {
        value: 'sss',
        parentElement: {},
      },
    };
    input.find('input').simulate('keyUp', event);
    expect(props.onKeyUp).toHaveBeenCalled();
  });

  test('should call keyup for remove', () => {
    const event = {
      target: { value: '', parentElement: {} },
    };
    input.find('input').simulate('change', event);
    expect(props.onChange).toHaveBeenCalled();
  });
});
