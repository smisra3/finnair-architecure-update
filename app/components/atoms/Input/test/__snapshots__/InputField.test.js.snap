// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render correctly 1`] = `
<div>
  <div
    className="label-wrapper"
  >
    <label />
  </div>
  <div
    className="input-wrapper"
  >
    <input
      aria-invalid={true}
      aria-required={null}
      data-slnm-id="test"
      dynamicPlaceholderLabel={true}
      hideLabel={true}
      iconClass="iconClass"
      isError={true}
      maxLength={1}
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder="test"
      type="text"
      valid={true}
      value=""
    />
  </div>
  <Styled(FieldError)
    dataSlnmId="InputField"
    errorMessage="errorMessage"
  />
</div>
`;

exports[`Input Component should render correctly 2`] = `
<div>
  <div
    className="label-wrapper"
  >
    <label />
  </div>
  <div
    className="input-wrapper"
  >
    <input
      aria-invalid={true}
      aria-required={null}
      data-slnm-id="test"
      dynamicPlaceholderLabel={true}
      hideLabel={false}
      iconClass="iconClass"
      isError={true}
      maxLength={1}
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder="test"
      type="text"
      valid={true}
      value=""
    />
  </div>
  <Styled(FieldError)
    dataSlnmId="InputField"
    errorMessage="errorMessage"
  />
</div>
`;

exports[`Input Component should render correctly 3`] = `
<div>
  <div
    className="label-wrapper"
  >
    <label />
  </div>
  <div
    className="input-wrapper"
  >
    <input
      aria-invalid={true}
      aria-required={null}
      data-slnm-id="test"
      dynamicPlaceholderLabel={true}
      hideLabel={false}
      isError={false}
      maxLength={1}
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder="test"
      type="text"
      valid={true}
      value=""
    />
  </div>
  <Styled(FieldError)
    dataSlnmId="InputField"
    errorMessage="errorMessage"
  />
</div>
`;
